### DISCUSSION TITLE
Automatically enable priviledged intents based on application flags

### DISCUSSION THREAD

**Discussion by davfsa** [hikari-py/hikari#1166] [Ideas]
Just a random thought I had. Would it be nice and worth it to make an extra call to fetch the bots application and enable privileged intents based on whether the flags for it are enabled or not (this would include also if they are pending) if no specific intents are passed to the bot constructor?

The downside I see for this is making a useless request a lot of the time, so I am a bit uncertain.

Would love to hear some thoughts :)

**Comment by hypergonial** [comment]
I think this could be a nice addition, with the condition that when the library does this, it emits a warning in the logs encouraging you to manually set your intents instead, and briefly explaining what is happening, maybe with a link to the Discord/hikari docs explaining intents.

**Comment by FasterSpeeding** [comment]
> it emits a warning in the logs encouraging you to manually set your intents instead

Bot devs don't tend to read library warnings, especially considering how many things may be logged on setup.

> this would include also if they are pending

Which would lead to the bot dying if it lost access to these intents without the bot's devs necessarily being aware that the bot was even currently declaring privileged intents.
